pandas/_libs/groupby.pyx

```diff
@@ -207,15 +207,24 @@ def group_cumprod_float64(
                         break
 
 
+ctypedef fused cumsum_t:
```

Comment:
What is the point of switching to this?

Reply:
Keeping precision for Int dtypes with missing values. Currently this is cast to float, which loses the value

also should help performance for ea arrays

Reply:
Also, this overflow pretty easily right now with int8 dtypes etc, this is also fixed

Reply:
Might be misreading but this is just a subset of the types that were previously used in `numeric_t` right? I'm a bit confused how restricting the types helps with performance / precision. Generally don't think we should be creating types specific to each algorithm

Reply:
There are two things here:

- When using int8 as a dtype, we can easily get overflows, because the type is not adjusted, for example a group with ``[111, 111]`` would overflow, so casting to ``int64`` beforehand avoids this. This is the bugfix

- Secondly, currently ea dtypes like `` Int64`` where cast to float before calling ``group_cumsum``, which lost precision for high integers that did not fit into ``float64``. Additionally, using the mask improves performance for extension array dtypes. 


I don't want to create specific dtypes per function. I want to keep the mask support for every function in separate prs. When more and more get merged, I will be able to combine the types and then we will be able to use these types for more functions.

Reply:
Renamed the type to make it clear that I intend to reuse it